### PR TITLE
fixes to mainnet docs

### PR DESCRIPTION
**Pull request by okwme**
also added the full and updated matrix for gaia versions (should stay in sync with version on roadmap)

### CODE REVIEW

File: docs/hub-tutorials/join-mainnet.md [hunk 1]
```diff
@@ -31,8 +31,8 @@ For instructions to join as a validator, please also see the [Validator Guide](h
     - [Blocksync](#blocksync)
     - [State Sync](#state-sync)
     - [Quicksync](#quicksync)
-- [Snapshots](#snapshots)=
```
Comment (glnro) [inline docs/hub-tutorials/join-mainnet.md:34]: Thanks for this

**Review by glnro — APPROVED** [review]

File: docs/hub-tutorials/join-mainnet.md [hunk 2]
```diff
@@ -261,12 +261,21 @@ For more information on performing the manual upgrades, see [Releases & Upgrades
...
+| Upgrade #2          | 07/08/20    | 2,902,000 | `cosmoshub-3` | [v0.32.x](https://github.com/tendermint/tendermint/releases/tag/v0.32.14)          | [v0.37.x](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.37.15)    | [v2.0.x](https://github.com/cosmos/gaia/releases/tag/v2.0.14)                   | n/a                      |
+| Stargate            | 18/02/21    | 5,200,791 | `cosmoshub-4` | [v0.34.x](https://github.com/tendermint/tendermint/releases/tag/v0.34.3)          | [v0.40.x](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.40.1)    | [v4.0.x](https://github.com/cosmos/gaia/releases/tag/v4.0.6)                   | _Included in Cosmos SDK_ |
+| Security Hard Fork  | ?             | ?         | `cosmoshub-4` | [v0.34.x](https://github.com/tendermint/tendermint/releases/tag/v0.34.8)          | [v0.41.x](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.41.4)    | [v4.2.x](https://github.com/cosmos/gaia/releases/tag/v4.2.1)                   | _Included in Cosmos SDK_ |
```
Comment (glnro) [inline docs/hub-tutorials/join-mainnet.md:272]: Should we leave this as TODO?

Comment (glnro) [reply docs/hub-tutorials/join-mainnet.md:272]: Looks like block `482100` https://forum.cosmos.network/t/critical-cosmossdk-security-advisory-updated/2211

Comment (okwme) [reply docs/hub-tutorials/join-mainnet.md:272]: looks like this was a missing rolling hardfork after mainnet launch and before upgrade #1. Added but still missing info from the 2021 hardfork. only think i could find about it was gaia release...

**Pull request merged**